### PR TITLE
Task/mov 678 api specifications

### PR DESCRIPTION
**Pull request by hb0**

### CODE REVIEW

File: persistence/src/main/java/de/cyface/persistence/serialization/MeasurementSerializer.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
  *
  * @author Klemens Muthmann
  * @author Armin Schnabel
- * @version 4.1.0
+ * @version 4.2.0
```
Comment (hb0) [inline persistence/src/main/java/de/cyface/persistence/serialization/MeasurementSerializer.java]: As this is a non-trivial  change (could break something somewhere) I decided to increase the 2nd number.

File: synchronization/src/androidTestMovebis/java/de/cyface/synchronization/DataTransmissionTest.java [hunk 2]
```diff
@@ -1,185 +0,0 @@
-/*
```
Comment (hb0) [inline synchronization/src/androidTestMovebis/java/de/cyface/synchronization/DataTransmissionTest.java]: This is renamed to SyncPerformerTest

File: datacapturing/src/cyface/java/de/cyface/datacapturing/CyfaceDataCapturingService.java [hunk 3]
```diff
@@ -71,17 +71,21 @@
...
      * @throws SetupException If writing the components preferences or registering the dummy user account fails.
      * @throws CursorIsNullException If {@link ContentProvider} was inaccessible.
      */
     private CyfaceDataCapturingService(@NonNull final Context context, @NonNull final ContentResolver resolver,
             @NonNull final String authority, @NonNull final String accountType,
             @NonNull final String dataUploadServerAddress, @NonNull final EventHandlingStrategy eventHandlingStrategy,
             @NonNull final DistanceCalculationStrategy distanceCalculationStrategy,
-            @NonNull final DataCapturingListener capturingListener) throws SetupException, CursorIsNullException {
+            @NonNull final DataCapturingListener capturingListener, final int sensorFrequency)
+            throws SetupException, CursorIsNullException {
```
Comment (muthenberg) [inline datacapturing/src/cyface/java/de/cyface/datacapturing/CyfaceDataCapturingService.java]: Warum sind denn die Exceptions plötzlich in einer neuen Zeile?

File: datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java [hunk 4]
```diff
@@ -205,14 +215,16 @@ public DataCapturingService(@NonNull final Context context, @NonNull final Strin
             @NonNull final EventHandlingStrategy eventHandlingStrategy,
             @NonNull final PersistenceLayer<CapturingPersistenceBehaviour> persistenceLayer,
             @NonNull final DistanceCalculationStrategy distanceCalculationStrategy,
-            @NonNull final DataCapturingListener capturingListener) throws SetupException, CursorIsNullException {
+            @NonNull final DataCapturingListener capturingListener, final int sensorFrequency)
+            throws SetupException, CursorIsNullException {
```
Comment (muthenberg) [inline datacapturing/src/main/java/de/cyface/datacapturing/DataCapturingService.java]: Hier auch wieder. ???

File: datacapturing/src/movebis/java/de/cyface/datacapturing/MovebisDataCapturingService.java [hunk 5]
```diff
@@ -124,17 +141,19 @@ public void onProviderDisabled(String provider) {
     public MovebisDataCapturingService(@NonNull final Context context, @NonNull final String dataUploadServerAddress,
             @NonNull final UIListener uiListener, final long locationUpdateRate,
             @NonNull final EventHandlingStrategy eventHandlingStrategy,
-            @NonNull final DataCapturingListener capturingListener) throws SetupException, CursorIsNullException {
+            @NonNull final DataCapturingListener capturingListener, final int sensorFrequency)
+            throws SetupException, CursorIsNullException {
```
Comment (muthenberg) [inline datacapturing/src/movebis/java/de/cyface/datacapturing/MovebisDataCapturingService.java]: und nochmal.

File: datacapturing/src/movebis/java/de/cyface/datacapturing/MovebisDataCapturingService.java [hunk 6]
```diff
@@ -159,11 +181,12 @@ public MovebisDataCapturingService(@NonNull final Context context, @NonNull fina
             @NonNull final String accountType, @NonNull final String dataUploadServerAddress,
             @NonNull final UIListener uiListener, final long locationUpdateRate,
             @NonNull final EventHandlingStrategy eventHandlingStrategy,
-            @NonNull final DataCapturingListener capturingListener) throws SetupException, CursorIsNullException {
+            @NonNull final DataCapturingListener capturingListener, final int sensorFrequency)
+            throws SetupException, CursorIsNullException {
```
Comment (muthenberg) [inline datacapturing/src/movebis/java/de/cyface/datacapturing/MovebisDataCapturingService.java]: again.

File: build.gradle [hunk 7]
```diff
@@ -44,6 +44,7 @@ ext {
 
     junitVersion = "4.12"
     mockitoVersion = "2.26.0"
+    //byteBuddyVersion = "1.9.10" // transitive dependency of mockito-code 2.26.0
```
Comment (muthenberg) [inline build.gradle]: Remove commented line?

File: synchronization/build.gradle [hunk 8]
```diff
@@ -88,6 +88,10 @@ dependencies {
     androidTestImplementation "androidx.test.ext:junit:$rootProject.ext.extJunitVersion"
     androidTestImplementation "androidx.test:rules:$rootProject.ext.rulesVersion"
     androidTestImplementation "org.mockito:mockito-android:$rootProject.ext.mockitoVersion"
+    // For some reasons the CI does does pull the transitive dependencies of mockito-android which will fails
+    // https://search.maven.org/remotecontent?filepath=org/mockito/mockito-android/2.26.0/mockito-android-2.26.0.pom
+    androidTestImplementation "org.mockito:mockito-core:$rootProject.ext.mockitoVersion"
+    //androidTestImplementation "net.bytebuddy:byte-buddy-android:$rootProject.ext.byteBuddyVersion"
```
Comment (muthenberg) [inline synchronization/build.gradle]: Remove the uncommented code?

File: build.gradle (hunk 7 above)
Comment (hb0) [reply build.gradle]: yep. was just waiting for the CI to cooperate before I remove this

**Pull request merged**